### PR TITLE
Added logic to save JUNIT log with ENV vars

### LINKED ISSUE #10508: enable junit reporting for extended test jobs
we should be optionally enabling the junit output in our extended test scripts like cmd.sh:

``` sh
# Allow setting $JUNIT_REPORT to toggle output behavior
if [[ -n "${JUNIT_REPORT:-}" ]]; then
  export JUNIT_REPORT_OUTPUT="${LOG_DIR}/raw_test_output.log"
fi
```

and then we should make sure the jenkins extended test jobs are settting JUNIT_REPORT as part of invoking the vagrant test action.

@oatmealraisin @stevekuznetsov

### PR DESCRIPTION
**Pull request by oatmealraisin**
Still need to configure Jenkins to use this. Copied the cleanup code from gssapi.sh

Fixes #10508

@bparees @stevekuznetsov ptal?

### CODE REVIEW

**Comment by bparees** [comment]
looks plausible to me, want to run some extended buckets against this to test it?

**Comment by stevekuznetsov** [comment]
Should wait for #11465

**Comment by oatmealraisin** [comment]
@stevekuznetsov Wait for #11465 to run tests? Or just merge?

@bparees Steve has configured the jenkins tests to set JUNIT_REPORT

**Comment by stevekuznetsov** [comment]
Wait for the other PR, after it hits you will have to re-work this one slightly.

**Comment by oatmealraisin** [comment]
Roger, dodger

**Comment by stevekuznetsov** [comment]
(you can get a leg up on it by making the changes, now, just see how that PR changed for instance `test-cmd.sh`)

**Comment by stevekuznetsov** [comment]
Blocker merged.

**Comment by oatmealraisin** [comment]
[test][testextended]

**Comment by stevekuznetsov** [comment]
[merge]

**Comment by oatmealraisin** [comment]
Flake #9886, [merge]

**Comment by stevekuznetsov** [comment]
@oatmealraisin you flake, I [merge]

**Comment by oatmealraisin** [comment]
@stevekuznetsov Was wondering how that would work w/ permissions. Noted

File: test/extended/ldap_groups.sh [hunk 1]
```diff
@@ -12,6 +12,27 @@ function cleanup()
...
+		# the middle of testing something
+		os::test::junit::reconcile_output
+
+		# check that we didn't mangle jUnit output
+		os::test::junit::check_test_counters
+
+		# use the junitreport tool to generate us a report
+		os::util::ensure::built_binary_exists 'junitreport'
+
+		cat "${JUNIT_REPORT_OUTPUT}" "${junit_gssapi_output}" \
```
Comment (liggitt) [inline test/extended/ldap_groups.sh]: extended run is failing with this:
```
test/extended/ldap_groups.sh: line 28: junit_gssapi_output: unbound variable
```

Comment (stevekuznetsov) [inline test/extended/ldap_groups.sh]: Fixed in master.

**Pull request merged**